src/grasshopper/lib/fixtures/__init__.py

```diff
@@ -531,3 +564,87 @@ def type_check_list_of_strs(list_of_strs):
             all_strs = all_strs and type(s) == str
         check_passed = all_strs
     return check_passed
+
+
+def _get_composite_weighted_user_classes(
+    full_scenarios_list, composite_scenario_contents
+):
+    """Generate a dictionary of journey classes with their associated weights."""
+    weighted_user_classes = {}
+    child_scenario_specs = _get_child_scenario_specs(
+        full_scenarios_list, composite_scenario_contents
+    )
+    for child_scenario_spec in child_scenario_specs:
+        base_journey_class = _import_class_with_journey(
+            absolute_file_path=f"{os.getcwd()}/"
+            f'{child_scenario_spec.get("test_file_name")}'
+        )
+        base_class_name = base_journey_class.__name__
+        composite_class_name = (
+            f"composite_journey_class_{base_class_name}_{uuid.uuid4()}"
+        )
+
+        # dynamically create a child class which inherits from the base class,
+        # required for having separate scenario args
+        child_journey_class = type(
+            composite_class_name,
+            (base_journey_class,),
+            {
+                "_incoming_test_parameters": child_scenario_spec.get(
+                    "grasshopper_scenario_args"
+                )
+            },
+        )
+        weighted_user_classes[child_journey_class] = child_scenario_spec.get("weight")
+    return weighted_user_classes
+
+
+def _import_class_with_journey(absolute_file_path):
+    """Import and return a class with 'journey' in its name from a module file."""
+    try:
+        module_name = os.path.splitext(absolute_file_path)[0]
+        spec = importlib.util.spec_from_file_location(module_name, absolute_file_path)
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+
+        # Now, inspect the module's attributes to find the class with "journey" in
+        # its name
+        for name, obj in vars(module).items():
+            if (
+                isinstance(obj, type)
+                and "journey" in name.lower()
+                and "base" not in name.lower()
+            ):
+                return obj
+
+        # If no class with "journey" in its name is found, return None or raise an
+        # exception
+        raise ImportError("No class with 'journey' in its name")
```

Comment:
this feels teeny bit confusing that you raise an error and then catch the error and do something. why not just return `None` here?